network/p2p/libp2pConnector.go

```diff
@@ -117,19 +118,32 @@ func (l *libp2pConnector) trimAllConnectionsExcept(peerInfos []peer.AddrInfo) {
 	allCurrentConns := l.host.Network().Conns()
 
 	// for each connection, check if that connection should be trimmed
+ConnectionLoop:
 	for _, conn := range allCurrentConns {
 
 		// get the remote peer ID for this connection
 		peerID := conn.RemotePeer()
 
 		// check if the peer ID is included in the current fanout
 		if peersToKeep[peerID] {
-			continue
+			continue ConnectionLoop
 		}
 
 		peerInfo := l.host.Network().Peerstore().PeerInfo(peerID)
 		log := l.log.With().Str("remote_peer", peerInfo.String()).Logger()
 
+		// retain the connection if there is a Flow One-to-One stream on that connection
+		// (we do not want to sever a connection with on going direct one-to-one traffic)
+		for _, s := range conn.GetStreams() {
+			streamProtocol := string(s.Protocol())
+			if strings.HasPrefix(streamProtocol, FlowLibP2PProtocolCommonPrefix) {
```

Comment:
if the stream protocol id is `/flow/push` or /flow/ping` then retain the connection

Reply:
doesn't this only check if it has `FlowLibP2PProtocolCommonPrefix`, which at the moment is `/flow`? i think we need to also check for `FlowLibP2POneToOneProtocolIDPrefix`

Reply:
so all Flow libp2p protocol will have this common prefix `/flow` hence I am only checking for that instead of checking for `/flow/push` and `/flow/ping`. The PubSub 1-to-k is not a Flow based protocol but instead something defined by libp2p as it is `/meshsub`.